tests/suites/test_suite_ctr_drbg.function

```diff
@@ -23,80 +23,29 @@ static int mbedtls_test_entropy_func( void *data, unsigned char *buf, size_t len
     test_offset_idx += len;
     return( 0 );
 }
-/* END_HEADER */
-
-/* BEGIN_DEPENDENCIES
- * depends_on:MBEDTLS_CTR_DRBG_C
- * END_DEPENDENCIES
- */
-
-/* BEGIN_CASE */
-void ctr_drbg_special_behaviours(  )
-{
-    mbedtls_ctr_drbg_context ctx;
-    unsigned char output[512];
-    unsigned char additional[512];
-
-    mbedtls_ctr_drbg_init( &ctx );
-    memset( output, 0, sizeof( output ) );
-    memset( additional, 0, sizeof( additional ) );
-
-    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx,
-                        output, MBEDTLS_CTR_DRBG_MAX_REQUEST + 1,
-                        additional, 16 ) ==
-                        MBEDTLS_ERR_CTR_DRBG_REQUEST_TOO_BIG );
-    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx,
-                        output, 16,
-                        additional, MBEDTLS_CTR_DRBG_MAX_INPUT + 1 ) ==
-                        MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG );
 
-    TEST_ASSERT( mbedtls_ctr_drbg_reseed( &ctx, additional,
-                        MBEDTLS_CTR_DRBG_MAX_SEED_INPUT + 1 ) ==
-                        MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG );
-
-    mbedtls_ctr_drbg_set_entropy_len( &ctx, ~0 );
-    TEST_ASSERT( mbedtls_ctr_drbg_reseed( &ctx, additional,
-                        MBEDTLS_CTR_DRBG_MAX_SEED_INPUT ) ==
-                        MBEDTLS_ERR_CTR_DRBG_INPUT_TOO_BIG );
-exit:
-    mbedtls_ctr_drbg_free( &ctx );
-}
-/* END_CASE */
-
-/* BEGIN_CASE */
-void ctr_drbg_validate( int reseed_mode, char *nonce_string,
-                        int entropy_len_arg, char *entropy_string,
-                        char *reseed_string,
-                        char *add1_string, char *add2_string,
-                        char *result_string )
+static int ctr_drbg_validate_internal( int reseed_mode, data_t * nonce,
+                        int entropy_len_arg, data_t * entropy,
+                        data_t * reseed,
+                        data_t * add1, data_t * add2,
+                        data_t * result )
 {
-    unsigned char entropy[144];
-    unsigned char nonce[64];
-    unsigned char reseed[32];
-    unsigned char add1[48];
-    unsigned char add2[48];
     mbedtls_ctr_drbg_context ctx;
     unsigned char buf[64];
-    unsigned char result[64];
+
     size_t entropy_chunk_len = (size_t) entropy_len_arg;
```

Comment:
Please add `TEST_ASSERT( entropy_chunk_len <= sizeof( buf ) )` so we don't get a nasty surprise later if we add a test with a different output size.

Reply:
check added